### PR TITLE
support path-based SourceFieldConfig

### PR DESCRIPTION
**Pull request by jaypipes**
Adds ability to specify a dot-notation object field path for the
`FieldConfig.From.Path` attribute. This allows us to specify nested
struct fields as the source field target, including for the Lambda
service controller which has some "funky" input and output shape
definitions.

To test the path-finding behaviour of the SourceFieldConfig, adds a new
`pkg/generate/lambda_test.go` file along with a `generator.yaml` for the
testdata-pinned Lambda api-2.json file that generates a CodeLocation and
CodeRepositoryType Status field from the GetFunction Output shape's
`Code.Location` and `Code.RepositoryType` members, respectively.

@A-Hilaly feel free to go to town on this :)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/model/sdk_helper.go [hunk 1]
```diff
@@ -168,35 +168,53 @@ func (a *SDKAPI) GetOperationMap(cfg *ackgenconfig.Config) *OperationMap {
...
-		if opID == operation && op.OutputRef.Shape != nil {
-			for opMemberName, memberShapeRef := range op.OutputRef.Shape.MemberRefs {
-				if opMemberName == memberName {
-					return memberShapeRef, true
-				}
-			}
+// getMemberByPath returns a ShapeRef given a root Shape and a dot-notation
+// object search path
+func getMemberByPath(
+	shape *awssdkmodel.Shape,
```
Comment (vijtrip2) [inline pkg/model/sdk_helper.go:200]: nit: Can mutating the value of the pointer have a side effect in future?
In the current scope, i do not see any harm.

File: pkg/model/sdk_helper_test.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+			true,
+		},
+		{
+			// nested path search
+			"GetFunction",
+			"Code.Location",
+			// Note that unlike CreateFunctionRequest.Description above, which
+			// is a Shape called `Description` that is itself a Go string type,
+			// the name of the `GetFunctionResponse.Code.Location` Shape is
+			// actually called `String`. Yep... this is why we can't have nice
```
Comment (vijtrip2) [inline pkg/model/sdk_helper_test.go:131]: lol

File: pkg/generate/lambda_test.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		"Tags",
+		"Timeout",
+		"TracingConfig",
+		"VPCConfig",
+	}
+	assert.Equal(expSpecFieldCamel, attrCamelNames(specFields))
+
+	expStatusFieldCamel := []string{
+		// Added from generator.yaml
+		"CodeLocation",
```
Comment (vijtrip2) [inline pkg/generate/lambda_test.go:82]: Just curious to know that the part where ReadOne output will populate "CodeLocation" with value of "Code.Location" is yet to be implemented. right?

**Review by vijtrip2 — APPROVED** [review]

File: pkg/model/sdk_helper.go (hunk 1 above)
Comment (jaypipes) [reply pkg/model/sdk_helper.go:200]: Yeah, most places in the code generator here pass pointers to either ShapeRef or Shape structs...

File: pkg/generate/lambda_test.go (hunk 3 above)
Comment (jaypipes) [reply pkg/generate/lambda_test.go:82]: Correct. @A-Hilaly was working on the Lambda service controller so I didn't want to step on his toes :) Just wanted to offer this generator config functionality to help him out.

**Review by A-Hilaly — COMMENTED** [review]
Sorry for the late review. LGTM! :+1:

File: pkg/generate/lambda_test.go (hunk 3 above)
Comment (A-Hilaly) [reply pkg/generate/lambda_test.go:82]: This will be very helpful! Thanks @jaypipes :)

**Pull request merged**